### ISSUE TITLE
Some testcases don't check if the sandbox is available before running causing test failure

### ISSUE THREAD

**Issue by matthewhilton** [trampgeek/moodle-qtype_coderunner#145]
AFAIK when any of the language tests run they should be checking if the language is available beforehand in the test environment:
https://github.com/trampgeek/moodle-qtype_coderunner/blob/db3a3d745a8df0b34cb0a019c9d91dde9f55c465/tests/test.php#L79-L84

For e.g. Octave questions check if the language is available:

https://github.com/trampgeek/moodle-qtype_coderunner/blob/db3a3d745a8df0b34cb0a019c9d91dde9f55c465/tests/octave_question_test.php#L42

However it seems some tests do not do this for e.g. `c_questions_testcase` :

https://github.com/trampgeek/moodle-qtype_coderunner/blob/db3a3d745a8df0b34cb0a019c9d91dde9f55c465/tests/c_questions_test.php#L40-L50

This means that servers that haven't setup the test config, but run all the tests will get an error instead of just skipped tests:

```
1) qtype_coderunner_c_questions_testcase::test_good_sqr_function
qtype_coderunner_exception: qtype_coderunner/No sandboxes available for running code!
$a contents: 

/var/www/site/question/type/coderunner/classes/sandbox.php:140
/var/www/site/question/type/coderunner/question.php:814
/var/www/site/question/type/coderunner/classes/jobrunner.php:63
/var/www/site/question/type/coderunner/question.php:554
/var/www/site/question/type/coderunner/tests/c_questions_test.php:43
/var/www/site/lib/phpunit/classes/advanced_testcase.php:80
phpvfscomposer:///var/www/site/vendor/phpunit/phpunit/phpunit:60

To re-run:
 vendor/bin/phpunit "qtype_coderunner_c_questions_testcase" question/type/coderunner/tests/c_questions_test.php
 ```

**Comment by trampgeek** [comment]
Thanks for pointing this out. It's now on my TODO list.

**Comment by trampgeek** [comment]
Closed, as problem fixed in version 5.2.4.
